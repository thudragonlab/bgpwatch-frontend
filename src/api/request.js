import axios from 'axios'
import {
  Message,
  MessageBox
} from 'element-ui'
// import { getToken } from '@/utils/auth'
// import { API_BASE_URL } from '@/utils/url_config'

// create an axios instance
let oldCrrentView = localStorage.getItem('currentView')
const service = axios.create({
  // baseURL: '', // url = base url + request url
  // withCredentials: true, // send cookies when cross-domain requests
  timeout: 30000 * 2
  // * 20, // request timeout
})

// request interceptor
service.interceptors.request.use(
  config => {
    // do something before request is sent
    // console.log(config)
    if (config.useToken) {
      const userInfo = localStorage.getItem('userInfo')
      if (userInfo) {
        const token = JSON.parse(userInfo).token
        // console.log(JSON.parse(userInfo).token)
        // config.headers['content-type'] = 'application/x-www-form-urlencoded;charset=UTF-8';
        config.headers['Authorization'] = 'Bearer ' + token
      }
    }

    return config
  },
  // error => {
  //   // do something with request error
  //   console.log(error) // for debug
  //   return Promise.reject(error)
  // }
)

// response interceptor
service.interceptors.response.use(
  /**
   * If you want to get http information such as headers or status
   * Please return  response => response
   */

  /**
   * Determine the request status by custom code
   * Here is just an example
   * You can also judge the status by HTTP Status Code
   */
  response => {
    const {
      data,
      headers
    } = response
    if (headers['content-type'] === "text/csv; charset=utf-8") {
      const fileName = headers['content-disposition'].replace(/\w+; filename=(.*)/, '$1')
      console.log(fileName)
      // 此处当返回json文件时需要先对data进行JSON.stringify处理，其他类型文件不用做处理
    } else {
      const data = response.data
      if(data.lastUpdateTimeStamp && data.lastUpdateTimeStamp > 0){
        const lastUpdateTimeStamp = data.lastUpdateTimeStamp
        const nowCrrentView = localStorage.getItem('currentView')
        if(oldCrrentView !== nowCrrentView){
          localStorage.setItem('lastUpdateTimestamp',lastUpdateTimeStamp)
          oldCrrentView = nowCrrentView
        }else{
          const oldLastUpdateTimestamp = localStorage.getItem('lastUpdateTimestamp')
          if(!oldLastUpdateTimestamp){
            localStorage.setItem('lastUpdateTimestamp',lastUpdateTimeStamp)  
          }else{
            const newUpdateTimestamp = Math.max(Number(lastUpdateTimeStamp),Number(oldLastUpdateTimestamp))
            localStorage.setItem('lastUpdateTimestamp',newUpdateTimestamp)
          }
        }
        
      }
      
      if (!data.status) {
        MessageBox({
          message: data.statusCode || 'Error',
          type: 'error',
          duration: 5 * 1000
        })
        return Promise.reject(data)
      }
    }
    return data.data
   
  },
  error => {
    console.error(error.message) // for debug
    if (error.message && error.message.includes('aborted')) {
      return Promise.reject(error)
    }

    MessageBox({
      message: 'No data found.',
      type: 'error',
      duration: 1 * 1000
    })
    return Promise.reject(error)
  }
)

export default service