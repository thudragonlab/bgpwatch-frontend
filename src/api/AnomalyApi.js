import request from './request'
const MY_URL = '/myapi'
export function getEventlistByDate(st, et) {
    return request({
        methods: 'GET',
        url: `${MY_URL}/events?st=${st}&et=${et}&event_type=ALL`,
        timeout: 3 * 60 * 1000,
    })
}
export function getMoasEventlistByDate(st, et) {
    return request({
        methods: 'GET',
        url: `${MY_URL}/events?st=${st}&et=${et}&event_type=ALL_MOAS`,
        timeout: 3 * 60 * 1000,
    })
}


export function getEventDetail(detail_url) {
    // return new Promise((resolve,reject) => {
    //   return resolve(eventsDetail)
    // })
    return request({
        methods: 'GET',
        url: `${MY_URL}${detail_url}`
    })
}


