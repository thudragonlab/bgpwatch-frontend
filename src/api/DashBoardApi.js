import request from './request'
import axios from 'axios'
// const BASE_URL = 'http://10.110.62.34:5000/'
const BASE_URL = '/dbapi'
const BASE_URL2 = '/newapi'
const MY_URL = '/myapi'
const WZQ_API2 = '/wzqApi2'


export function getAsInfoByASN(asn) {
  return request({
    method: 'GET',
    url: `${BASE_URL}/api/dashboard/daily/asinfo?asn=${asn}`,

  })
}

export function getAsInfoByASN_myAPI(asn) {
  return request({
    method: 'GET',
    url: `${MY_URL}/asInfo?asn=${asn}`,

  })
}


export function getPeerPrefix(asn, {
  type,
  detailAsn
}) {
  return request({
    method: 'GET',
    url: `${BASE_URL}/api/dashboard/daily/peerprefix?asn=${asn}&${type}=${detailAsn}`,
  })
}



export function getAsRoute(asn, type) {
  return request({
    method: 'GET',
    url: `${MY_URL}/getAsRoute?asn=${asn}&type=${type}`,
  })
}

export function getPeerTypeDistribute(asn) {
  return request({
    method: 'GET',
    url: `${MY_URL}/getPeerTypeDistribute?asn=${asn}`,
  })
}


export function getAsInfoByList(asnList) {
  return request({
    method: 'POST',
    url: `${MY_URL}/getAsInfoByList`,
    data: asnList.map(as => parseInt(as))
  })
}

export function getRealTimeData() {
  return request({
    method: 'GET',
    url: `${MY_URL}/getRealTimeData`,
  })
}

export function getRealTimeDataInAWhile() {
  return request({
    method: 'GET',
    url: `${MY_URL}/getRealTimeDataInAWhile`,
  })
}

export function getASList(asn) {
  return request({
    method: 'GET',
    url: `${MY_URL}/getAsList?asn=${asn}`,
  })
}

// import test from '@/../public/static/temporaryData/1.json'
export function getAsPathByPrefix(prefix, asn,type) {
  // return new Promise((resolve,reject) => {
  //   resolve(test)
  // })
  return request({
    method: 'GET',
    url: `${MY_URL}/getAsPathByPrefix?prefix=${prefix}&asn=${asn}&type=${type}`,
  })
}


export function getRouteHistoryStats(asn, qtype = 'g') {
  return request({
    method: 'GET',
    url: `${MY_URL}/route_history_stats?asns=${asn.join(',')}&qtype=${qtype}`,
  })
}


export function getRoutingPath(asn, pfx, qtype = "from") {
  return request({
    method: 'GET',
    url: `${WZQ_API2}/api/routing_path?asn=${asn}&pfx=${pfx}&qtype=${qtype}`
  })
}


export function getPrefixList(asn, qstr = undefined) {
  return request({
    method: 'GET',
    url: `${MY_URL}/prefix_list?asn=${asn}${qstr ? `&qstr=${qstr}` :  ''}`
  })
}

export function getPrefixTotalByAsn(asn) {
  return request({
    method: 'GET',
    url: `${MY_URL}/getPrefixTotalByAsn?asn=${asn}`,
  })
}

export function getPeerList(asn,ipVersion) {
  return request({
    method: 'GET',
    url: `${MY_URL}/peer_list?asn=${asn}&ipversion=${ipVersion}`
  })
}


// http://10.110.62.37:5050/api/routing_path?asn=4538&pfx=59.66.196.0%2F24&qtype=to

export function getWhoisInfoByASN(asn){
  return request({
    method: 'GET',
    url: `${MY_URL}/get_whois_info_by_asn?asn=${asn}`
  })
}