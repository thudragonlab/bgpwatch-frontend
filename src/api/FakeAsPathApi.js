import request from './request'
const MY_URL = '/myapi'

export function getFakeAsPath(st, et) {
    return request({
        methods: 'GET',
        url: `${MY_URL}/getFakeAsPath?st=${st}&et=${et}`,
        timeout: 3 * 60 * 1000,
    })
}


export function getFakeAsPathRecordById(id){

    return request({
        methods: 'GET',
        url: `${MY_URL}/getFakeAsPathRecordById?id=${id}`,
        timeout: 3 * 60 * 1000,
    })

}