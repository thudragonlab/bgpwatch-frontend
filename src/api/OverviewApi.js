import request from './request'
import axios from 'axios'
// const BASE_URL = 'http://10.110.62.34:5000/'
const MY_URL = '/myapi'


export function getOverviewData(startTime, endTime,selectedOption,asn) {
    return request({
      methods: 'GET',
      url: `${MY_URL}/overview?start=${startTime}&end=${endTime}&cc=${selectedOption}&asn=${asn}`,
    })
  }


export function downloadCSV(startTime, endTime) {
    return request({
        methods: 'GET',
        url: `${MY_URL}/exportCSV?start=${startTime}&end=${endTime}`,
        responseType:'blob'
      })
}