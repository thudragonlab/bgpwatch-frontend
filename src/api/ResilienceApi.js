import request from './request'
const MY_URL = '/myapi'

export function getResilienceByccCode(cc) {
  return request({
    method: 'GET',
    url: `${MY_URL}/getResilienceByccCode?cc=${cc}`,
  })
}


export function getResilienceTestData(cc, asn) {
  return request({
    method: 'GET',
    url: `${MY_URL}/getTopologyData?cc=${cc}&asn=${asn}`,
  })
}


export function getResilienceTestDataInOptimize(cc, asn) {
  return request({
    method: 'GET',
    url: `${MY_URL}/getTopologyDataOpt?cc=${cc}&asn=${asn}`,
  })
}

export function getAsByCc(cc) {
  return request({
    method: 'GET',
    url: `${MY_URL}/getAsByCc?cc=${cc}`,
  })
}


export function getRtreeByAsn(cc, asn, condition) {
  return request({
    method: 'GET',
    url: `${MY_URL}/getRtreeByAsn?cc=${cc}&asn=${asn}&condition=${condition}`,
  })
}

