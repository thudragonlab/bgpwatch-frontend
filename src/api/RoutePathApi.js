import request from './request'
const MY_URL = '/myapi'

export function getRtreeByPrefix(prefix){
  return request({
    method: 'GET',
    url: `${MY_URL}/getRtreeByPrefix?prefix=${prefix}`,
  })
}

export function getRtreeByPrefixData(prefix){
  return request({
    method: 'GET',
    url: `${MY_URL}/getRtreeByPrefixData?prefix=${prefix}`,
  })
}

export function getRtreeByPrefixDataTopo(prefix){
  return request({
    method: 'GET',
    url: `${MY_URL}/getRtreeByPrefixDataToTopo?prefix=${prefix}`,
  })
}

export function getRouteByIP(left_ip,right_ip){
  return request({
    method: 'GET',
    url: `${MY_URL}/getRouteByIP?left_ip=${left_ip}&right_ip=${right_ip}`,
  })
}

export function getRoutingPathCluster(asn){
  return request({
    method: 'GET',
    url: `${MY_URL}/getRoutingPathCluster?asn=${asn}`,
  })
}

export function getRtreeByHash(hash){
  return request({
    method: 'GET',
    url: `${MY_URL}/getRtreeByHash?hash=${hash}`,
  })
}


export function getRtreeByHashes(hashes,asn){
  return request({
    method: 'POST',
    url: `${MY_URL}/getRtreeByHashes?asn=${asn}`,
    data:hashes
  })
}


export function getJitterTopNPrefixByAs(asn,st,et,source){
  return request({
    method: 'GET',
    url: `${MY_URL}/getJitterTopNPrefixByAs?asn=${asn}&st=${st}&et=${et}&source=${source}`,
  })
}

export function getJitterTopNPrefixByAsAndPfx(asn,prefix,st,et,source){
  return request({
    method: 'GET',
    url: `${MY_URL}/getJitterTopNPrefixByAsAndPfx?asn=${asn}&prefix=${prefix}&st=${st}&et=${et}&source=${source}`,
  })
}



export function getJitterDataByPrefix(asn,prefix,st,et,source){
  return request({
    method: 'GET',
    url: `${MY_URL}/getJitterDataByPrefix?prefix=${prefix}&asn=${asn}&st=${st}&et=${et}&source=${source}`,
  })
}

export function getJitterDataByASNAndPfx(asn,ppfx,prefix,st,et,source){
  return request({
    method: 'GET',
    url: `${MY_URL}/getJitterDataByASNAndPfx?prefix=${prefix}&ppfx=${ppfx}&asn=${asn}&st=${st}&et=${et}&source=${source}`,
  })
}



export function getFITIJitterTopNPrefixByPrefix(prefix,st,et,source){
  return request({
    method: 'GET',
    url: `${MY_URL}/getFITIJitterTopNPrefixByPrefix?prefix=${prefix}&st=${st}&et=${et}&source=${source}`,
  })
}

export function getFITIJitterDataByPrefix(p_prefix,prefix,st,et,source){
  return request({
    method: 'GET',
    url: `${MY_URL}/getFITIJitterDataByPrefix?prefix=${prefix}&p_prefix=${p_prefix}&st=${st}&et=${et}&source=${source}`,
  })
}

export function getJitterTopNPeerByPfx(asn,pfx,st,et,source){
  return request({
    method: 'GET',
    url: `${MY_URL}/getJitterTopNPeerByPfx?asn=${asn}&pfx=${pfx}&st=${st}&et=${et}&source=${source}`,
  })
}

export function getJitterTopNPrefixByPeer(asn,st,et,source){
  return request({
    method: 'GET',
    url: `${MY_URL}/getJitterTopPeer?asn=${asn}&st=${st}&et=${et}&source=${source}`,
  })
}


export function getJitterDataByPeer(asn,peer,st,et,source){
  return request({
    method: 'GET',
    url: `${MY_URL}/getJitterDataByPeer?asn=${asn}&peer=${peer}&st=${st}&et=${et}&source=${source}`,
  })
}

export function getJitterDataByPeerAndPfx(asn,pfx,peer,st,et,source){
  return request({
    method: 'GET',
    url: `${MY_URL}/getJitterDataByPeerAndPfx?asn=${asn}&pfx=${pfx}&peer=${peer}&st=${st}&et=${et}&source=${source}`,
  })
}


export function getJitterPeerASPathByTimeStamp(asn,peer,ts,page,timeFrame,source){
  return request({
    method: 'GET',
    url: `${MY_URL}/getJitterPeerASPathByTimeStamp?asn=${asn}&ts=${ts}&peer=${peer}&page=${page}&time_f=${timeFrame}&source=${source}`,
  })
}

