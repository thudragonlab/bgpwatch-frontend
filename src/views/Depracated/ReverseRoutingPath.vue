<template>
  <div class="reverse-route-path-container fade-in">
    <div class="prefix-input">
      <div>
        <el-input placeholder="1.0.0.0" v-model="inputPrefix">
          <template slot="prepend">Prefix</template>
          <el-button slot="append" icon="el-icon-search" @click="searchPrefix"></el-button>
        </el-input>
      </div>
      <div style="text-align: left; padding-right: 20px">
        <div>You can input an IP or prefix, such as 1.0.205.0, 1.0.205.0/24 or 2001:200::/32.</div>
        <div>The system will search the best matched prefix and return the reverse routing tree.</div>
      </div>
    </div>
    <el-row type="flex" justify="space-between">
      <div v-if="rtree_images.length && rtreeBase64Code != 'Error'" class="rtree-image" v-viewer="{ movable: false }">
        <!-- <div class="title show-row">intra-region routing tree {{ rtree_condition }}</div> -->
        <img
          class="show-row"
          v-for="(src, index) in rtree_images"
          :src="src"
          :key="index"
          style="width: auto; height: 100%"
        />
      </div>
      <div v-if="rtreeBase64Code == 'Error'" class="not-found">NOT FOUND</div>
    </el-row>
  </div>
</template>

<script>
import { getRtreeByPrefix } from '@/api/RoutePathApi.js'
const IP_REGEX = /(^(1\d{2}|2[0-4]\d|25[0-5]|[1-9]\d|[0-9])\.((1\d{2}|2[0-4]\d|25[0-5]|[1-9]\d|\d)\.){2}(1\d{2}|2[0-4]\d|25[0-5]|[1-9]\d|\d)$)|(^(1\d{2}|2[0-4]\d|25[0-5]|[1-9]\d|[0-9])\.((1\d{2}|2[0-4]\d|25[0-5]|[1-9]\d|\d)\.){2}(1\d{2}|2[0-4]\d|25[0-5]|[1-9]\d|\d)\/(1[0-9]|[1-9]|2[0-4])$)|(\s*((([0-9A-Fa-f]{1,4}:){7}([0-9A-Fa-f]{1,4}|:))|(([0-9A-Fa-f]{1,4}:){6}(:[0-9A-Fa-f]{1,4}|((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3})|:))|(([0-9A-Fa-f]{1,4}:){5}(((:[0-9A-Fa-f]{1,4}){1,2})|:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3})|:))|(([0-9A-Fa-f]{1,4}:){4}(((:[0-9A-Fa-f]{1,4}){1,3})|((:[0-9A-Fa-f]{1,4})?:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){3}(((:[0-9A-Fa-f]{1,4}){1,4})|((:[0-9A-Fa-f]{1,4}){0,2}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){2}(((:[0-9A-Fa-f]{1,4}){1,5})|((:[0-9A-Fa-f]{1,4}){0,3}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){1}(((:[0-9A-Fa-f]{1,4}){1,6})|((:[0-9A-Fa-f]{1,4}){0,4}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(:(((:[0-9A-Fa-f]{1,4}){1,7})|((:[0-9A-Fa-f]{1,4}){0,5}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:)))(%.+)?\s*)/
// import BaseD3Graph from '@/components/RoutMonitor/BaseD3Graph.vue'
export default {
  name: 'ReverseRoutingPath',
  created() {},
  data() {
    return {
      rtree_images: [],
      rtreeBase64Code: '',
      inputPrefix: '1.0.128.0/17',
    }
  },
  components: {},
  props: {
    param: {
      type: Object,
      default: () => {
        return {}
      },
    },
    inputAsn: {
      type: String,
      default: '',
    },
  },
  watch: {},
  mounted() {
    this.$nextTick(() => {
      this.getRtreeByPrefix().catch(e => console.error(e))
    })
  },
  methods: {
    async getRtreeByPrefix() {
      const rtreeData = await getRtreeByPrefix(this.inputPrefix)
      if (!rtreeData) {
        this.rtreeBase64Code = ''
        this.rtree_images = []
        return
      }
      const { base64_code } = rtreeData
      if (base64_code === 'Error') {
        this.$messageBox({
          message: rtreeData.msg,
          type: 'error',
          duration: 2 * 1000,
        })
        this.rtreeBase64Code = 'Error'
        this.rtree_images = []
      } else {
        this.rtreeBase64Code = base64_code
        this.rtree_images = [base64_code]
      }
    },
    async searchPrefix() {
      if (!IP_REGEX.test(this.inputPrefix)) {
        this.$messageBox({
          message: `IP invalid`,
          type: 'warning',
          duration: 2 * 1000,
        })
        return
      }
      await this.getRtreeByPrefix().catch(e => console.error(e))
    },
  },
}
</script>

<style lang="scss" scoped>
.reverse-route-path-container {
  // padding-top:20px;
  background-color: #f0f0f0;
  height: 100%;
}

.prefix-input {
  width: 100%;
  padding: 20px 0 20px 0;
  display: flex;
  justify-content: space-between;
  align-items: center;
}
.rtree-image {
  height: 60vh;
  overflow: hidden;
  overflow-x: auto;
}

.not-found {
  height: 20vh;
  margin: 10px;
  width: calc(100% - 10px);
  line-height: 20vh;
  text-align: center;
  font-size: var(--max-font-size);
  font-weight: bold;
  background-color: #fff;
  border: 5px solid #000;
}
</style>
