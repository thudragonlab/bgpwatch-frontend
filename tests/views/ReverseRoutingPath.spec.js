import {
  shallowMount,
} from "@vue/test-utils";
import ElementUI from '@/lib/element.js'
import ReverseRoutingPath from "@/views/Depracated/ReverseRoutingPath";
import "jest-canvas-mock";
import { getRtreeByPrefix } from '@/api/RoutePathApi'
import Vue from "vue";
// import Viewer from 'v-viewer'
jest.mock('@/api/RoutePathApi', () => {
  return {
    getRtreeByPrefix: jest.fn(() => Promise.resolve({base64_code:''}))
  }
})

// Viewer.setDefaults()
// Vue.use(Viewer)

Vue.use(ElementUI)
Vue.prototype.$message = jest.fn()

describe.skip('ReverseRoutingPath.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(ReverseRoutingPath)
  })
  afterEach(() => {
    jest.clearAllMocks()
  })
  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('should call getRtreeByPrefix when searchPrefix is called with valid IP', async () => {
    wrapper.setData({ inputPrefix: '1.0.128.0/17' })
    getRtreeByPrefix.mockResolvedValueOnce({ base64_code: 'base64_code' })
    await wrapper.vm.searchPrefix()
    expect(getRtreeByPrefix).toBeCalledWith('1.0.128.0/17')
    expect(wrapper.vm.rtreeBase64Code).toBe('base64_code')
    expect(wrapper.vm.rtree_images).toEqual(['base64_code'])
  })
  it('should not call getRtreeByPrefix when searchPrefix is called with invalid IP', async () => {
    getRtreeByPrefix.mockReset()
    wrapper.setData({ inputPrefix: 'invalid_ip' })
    await wrapper.vm.searchPrefix()
    expect(wrapper.vm.$message).toHaveBeenCalledWith({
      message: 'IP invalid',
      type: 'warning',
      duration: 2000,
    })
    expect(getRtreeByPrefix).not.toHaveBeenCalled()
    expect(wrapper.vm.rtreeBase64Code).toBe('')
    expect(wrapper.vm.rtree_images).toEqual([""])

    getRtreeByPrefix.mockResolvedValueOnce(undefined)
    wrapper.setData({ inputPrefix: '1.0.128.0/17' })
    await wrapper.vm.searchPrefix()
    expect(getRtreeByPrefix).toHaveBeenCalled()
    expect(wrapper.vm.rtreeBase64Code).toBe('')
    expect(wrapper.vm.rtree_images).toEqual([])
    
  })
  it('should show error message when getRtreeByPrefix fails', async () => {
    wrapper.setData({ inputPrefix: '1.0.128.0/17' })
    const errorMessage = {base64_code:'Error',msg:'Error Message'}
    getRtreeByPrefix.mockResolvedValueOnce(errorMessage)
    await wrapper.vm.searchPrefix()
    expect(wrapper.vm.$message).toHaveBeenCalledWith({
      message: errorMessage.msg,
      type: 'error',
      duration: 2000,
    })
    expect(getRtreeByPrefix).toBeCalledWith('1.0.128.0/17')
    expect(wrapper.vm.rtreeBase64Code).toBe('Error')
    expect(wrapper.vm.rtree_images).toEqual([])
  })
})
