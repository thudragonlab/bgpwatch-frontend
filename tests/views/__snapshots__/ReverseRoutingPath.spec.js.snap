// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReverseRoutingPath.vue renders correctly 1`] = `
<div class="reverse-route-path-container fade-in">
  <div class="prefix-input">
    <div>
      <el-input-stub value="1.0.128.0/17" type="text" validateevent="true" placeholder="1.0.0.0"><template></template>
        <el-button-stub type="default" icon="el-icon-search" nativetype="button"></el-button-stub>
      </el-input-stub>
    </div>
    <div style="text-align: left; padding-right: 20px;">
      <div>You can input an IP or prefix, such as 1.0.205.0, 1.0.205.0/24 or 2001:200::/32.</div>
      <div>The system will search the best matched prefix and return the reverse routing tree.</div>
    </div>
  </div>
  <el-row-stub tag="div" type="flex" justify="space-between">
    <div class="rtree-image"><img src="" class="show-row" style="width: auto; height: 100%;"></div>
    <!---->
  </el-row-stub>
</div>
`;
